### PR TITLE
ci: update release steps

### PR DESCRIPTION
**Pull request by honestsubhadip**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

* update semantic release steps

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-gke/93)

### CODE REVIEW

File: .github/workflows/terratest.yaml [hunk 1]
```diff
@@ -43,11 +43,12 @@ jobs:
...
-        uses: cycjimmy/semantic-release-action@v2
+        uses: cycjimmy/semantic-release-action@v3
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
+          semantic_version: 18
           extra_plugins: |
             @semantic-release/git@10.0.1
-            @semantic-release/exec@6.0.3
+            @semantic-release/exec@6.0.2
```
Comment (MXfive) [inline .github/workflows/terratest.yaml:53]: Downgrade?

Comment (honestsubhadip) [reply .github/workflows/terratest.yaml]: ah my bad, copied from old repo.

**Review by jai — CHANGES_REQUESTED** [review]
incorrect PR type - please see https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c?pvs=4


Also, token permissions probably won't work - semantic release needs `content: write` and `pull-requests: write` - see https://github.com/honestbank/terraform-pagerduty-alerting/blob/main/.github/workflows/terratest.yaml

**Comment by honestsubhadip** [comment]
> incorrect PR type - please see https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c?pvs=4
> 
> Also, token permissions probably won't work - semantic release needs `content: write` and `pull-requests: write` - see https://github.com/honestbank/terraform-pagerduty-alerting/blob/main/.github/workflows/terratest.yaml

@jai I see, thanks for the doc and pointing the permission.

**Review by MXfive — APPROVED** [review]

File: .github/workflows/terratest.yaml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 name: terratest
-permissions: read-all
+permissions:
+  content: write
```
Comment (jai) [inline .github/workflows/terratest.yaml:3]: please check https://docs.github.com/en/actions/security-guides/automatic-token-authentication

Comment (jai) [inline .github/workflows/terratest.yaml:3]:
```suggestion
  contents: write
```

**Review by MXfive — APPROVED** [review]

**Pull request closed without merging**